src/core/qgslegendrenderer.cpp

```diff
@@ -661,7 +676,18 @@ QSizeF QgsLegendRenderer::drawGroup( const LegendComponentGroup &group, QgsRende
           currentY += mSettings.style( s ).margin( QgsLegendStyle::Top );
         }
         QSizeF groupSize;
-        groupSize = drawGroupTitle( groupItem, context, columnContext, currentY );
+        ColumnContext columnContextForItem = columnContext;
+        double indentWidth =  component.indent;
+        if ( s == QgsLegendStyle::Subgroup )
+        {
+          indentWidth -= mSettings.style( QgsLegendStyle::Subgroup ).indent( );
+        }
+        else
+        {
+          indentWidth -= mSettings.style( QgsLegendStyle::Group ).indent( );
+        }
+        columnContextForItem.left += indentWidth;
```

Comment:
For right-to-left legend arrangements this would need to be applied to the right side of the column

Reply:
This is now implemented